src/store/workflows.module.js

```diff
@@ -40,6 +40,28 @@ function _getCycles (workflows) {
   return cycles
 }
 
+/**
+ * Compute percent progress.
+ *
+ * @see https://github.com/cylc/cylc-flow/blob/de7d938496e82dbdfb165938145670dd8e801efd/lib/cylc/gui/updater_tree.py#L248-L263
+ * @param {number} startedTime in milliseconds since 1970-01-01 00:00:00 UTC, e.g. 1568353099874
+ * @param {number} meanElapsedTime mean elapsed time (e.g. 13.5)
+ * @returns {number} the percent progress, e.g. 25 (meaning 25% progress)
+ * @private
+ */
+function _computePercentProgress (startedTime, meanElapsedTime) {
+  const now = Date.now()
+  // startedTime > now reportably possible via interaction with `cylc reset`
+  if (startedTime > now || meanElapsedTime === 0) {
+    return 0
+  }
+  const elapsedTime = startedTime + meanElapsedTime
+  if (now > elapsedTime) {
+    return 100
+  }
+  return 100 * (now - startedTime) / meanElapsedTime
```

Comment:
Looks like it would work!

Reply:
Yeah, I changed very little the logic, only because this function is returning the value. But from what I could tell, it was supposed to give the same result. Just finishing some performance test on the complex suite + Cylc UI, then will jump back to this one.